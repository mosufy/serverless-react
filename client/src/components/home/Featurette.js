import React, { Component } from 'react';

class Featurette extends Component {
  render() {
    let {imgSrc, pullDirection, heading, subHeading, text} = this.props;

    return (
      <div className="featurette">
        <img className={`featurette-image img-circle img-responsive pull-${pullDirection}`} src={imgSrc} alt=""/>
        <h2 className="featurette-heading">{heading} <span className="text-muted">{subHeading}</span></h2>
        <p className="lead">{text}</p>
      </div>
    );
  }
}

export default Featurette;