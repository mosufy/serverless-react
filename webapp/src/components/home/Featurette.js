import React, { Component } from 'react';
import { Button } from 'react-bootstrap';

class Featurette extends Component {
  render() {
    let {imgSrc, pullDirection, heading, subHeading, text, handleButtonClick} = this.props;

    return (
      <div className="featurette">
        <img className={`featurette-image img-circle img-responsive pull-${pullDirection}`} src={imgSrc} alt=""/>
        <h2 className="featurette-heading">{heading} <span className="text-muted">{subHeading}</span></h2>
        <p className="lead">{text}</p>
        <Button bsStyle="danger" bsSize="large" onClick={handleButtonClick}>Register Interest</Button>
      </div>
    );
  }
}

export default Featurette;