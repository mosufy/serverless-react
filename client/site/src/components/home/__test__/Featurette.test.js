import React from 'react';
import renderer from 'react-test-renderer';

import Featurette from '../Featurette';

describe('home.Featurette component testing', () => {
  it('it renders correctly', () => {
    const tree = renderer.create(
      <Featurette />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});