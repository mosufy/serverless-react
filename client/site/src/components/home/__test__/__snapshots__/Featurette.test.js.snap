// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home.Featurette component testing it renders correctly 1`] = `
<div
  className="featurette"
>
  <img
    alt=""
    className="featurette-image img-circle img-responsive pull-undefined"
    src={undefined}
  />
  <h2
    className="featurette-heading"
  >
     
    <span
      className="text-muted"
    />
  </h2>
  <p
    className="lead"
  />
  <button
    className="btn btn-lg btn-danger"
    disabled={false}
    onClick={undefined}
    type="button"
  >
    Register Interest
  </button>
</div>
`;
